File: app/src/main/java/com/ericho/coupleshare/mvp/PhotosContract.java
package com.ericho.coupleshare.mvp;

import java.util.List;

/**
 * Created by steve_000 on 17年6月25日.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */

public interface PhotosContract {
    interface View extends BaseView<Presenter>{
        void showAddPhotoUI();

        void showPhotoList(List<Photo> photos);

    }
    interface Presenter extends BasePresenter{
        void loadPhotos();

        void addNewPhotos();
    }
}

File: app/src/main/java/com/ericho/coupleshare/mvp/PhotosContract.kt
package com.ericho.coupleshare.mvp

/**
 * Created by steve_000 on 6/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */

interface PhotosContract{
    interface View : BaseView<Presenter> {
        fun showAddPhotoUI()

        fun showPhotoList(photos: List<PhotoBo>)

    }

    interface Presenter : BasePresenter {
        fun loadPhotos()

        fun addNewPhotos()
    }
}